lib/cylc/task_events_mgr.py

```diff
@@ -346,43 +351,46 @@ def process_message(self, itask, severity, message, poll_func,
             message=message, is_completed=True)
 
         if message == TASK_OUTPUT_STARTED:
-            if self._poll_to_confirm(itask, TASK_STATUS_RUNNING, poll_func):
-                return
+            if (flag == self.INCOMING_FLAG
+                    and itask.state.is_gt(TASK_STATUS_RUNNING)):
+                return True
             self._process_message_started(itask, event_time)
         elif message == TASK_OUTPUT_SUCCEEDED:
-            if self._poll_to_confirm(itask, TASK_STATUS_SUCCEEDED, poll_func):
-                return
             self._process_message_succeeded(itask, event_time)
         elif message == TASK_OUTPUT_FAILED:
-            if self._poll_to_confirm(itask, TASK_STATUS_FAILED, poll_func):
-                return
+            if (flag == self.INCOMING_FLAG
+                    and itask.state.is_gt(TASK_STATUS_FAILED)):
+                return True
             self._process_message_failed(itask, event_time, self.JOB_FAILED)
         elif message == self.EVENT_SUBMIT_FAILED:
-            if self._poll_to_confirm(itask,
-                                     TASK_STATUS_SUBMIT_FAILED, poll_func):
-                return
+            if (flag == self.INCOMING_FLAG
+                    and itask.state.is_gt(TASK_STATUS_SUBMIT_FAILED)):
+                return True
             self._process_message_submit_failed(itask, event_time)
         elif message == TASK_OUTPUT_SUBMITTED:
-            if self._poll_to_confirm(itask, TASK_STATUS_SUBMITTED, poll_func):
-                return
+            if (flag == self.INCOMING_FLAG
+                    and itask.state.is_gt(TASK_STATUS_SUBMITTED)):
+                return True
             self._process_message_submitted(itask, event_time)
         elif message.startswith(FAIL_MESSAGE_PREFIX):
             # Task received signal.
             signal = message[len(FAIL_MESSAGE_PREFIX):]
             self._db_events_insert(itask, "signaled", signal)
             self.suite_db_mgr.put_update_task_jobs(
                 itask, {"run_signal": signal})
-            if self._poll_to_confirm(itask, TASK_STATUS_FAILED, poll_func):
-                return
+            if (flag == self.INCOMING_FLAG
```

Comment:
Why does this line not come immediately after `elif message.startswith(FAIL_MESSAGE_PREFIX):`?

Reply:
Logic updated.